proc_macros/src/lib.rs

```diff
@@ -165,3 +203,599 @@ pub fn rule_tests(input: TokenStream) -> TokenStream {
     }
     .into()
 }
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
```

Comment:
For a taste of what this stuff is I'd jump ahead and look at the `rule!()` invocation in `src/tests/fixing.rs` that uses some rule "state" fields